Lib/pathlib.py

```diff
@@ -920,12 +920,26 @@ def relative_to(self, *other):
             to_abs_parts = to_parts
         n = len(to_abs_parts)
         cf = self._flavour.casefold_parts
-        if (root or drv) if n == 0 else cf(abs_parts[:n]) != cf(to_abs_parts):
+        common = 0
+        for p, tp in zip(cf(abs_parts), cf(to_abs_parts)):
+            if p != tp:
+                break
+            common += 1
+        if strict:
+            failure = (root or drv) if n == 0 else common != n
+            error_message = "{!r} does not start with {!r}"
+            up_parts = []
+        else:
+            failure = root != to_root
+            if drv or to_drv:
+                failure = cf([drv]) != cf([to_drv]) or (failure and n > 1)
+            error_message = "{!r} is not related to {!r}"
+            up_parts = (n-common)*['..']
+        if failure:
             formatted = self._format_parsed_parts(to_drv, to_root, to_parts)
-            raise ValueError("{!r} does not start with {!r}"
-                             .format(str(self), str(formatted)))
-        return self._from_parsed_parts('', root if n == 1 else '',
-                                       abs_parts[n:])
+            raise ValueError(error_message.format(str(self), str(formatted)))
```

Comment:
How about using f-strings?

Reply:
The actual message is formatted only if the exception occurs and it needs a value prepared only in that case; so it can be done, but I think it would be a tittle wasteful.